File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerDisconnectEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;
import top.likoslupus.cellulosesz.api.teleport.CellLocation;

import static java.util.Objects.requireNonNull;

public record PlayerDisconnectEvent(
        CellPlayer player,
        CellLocation location
) {

    public PlayerDisconnectEvent {
        requireNonNull(player, "player");
        requireNonNull(location, "location");
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerDisconnectEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import top.likoslupus.cellulosesz.api.teleport.CellLocation

@JvmRecord
public data class PlayerDisconnectEvent(
    public val player: CellPlayer,
    public val location: CellLocation
)
